AssertError in tests.config.TestControlledDict when run on Python 3.11

**Describe the bug**

AssertError in tests.config.TestControlledDict when run on Fedora Rawhide(37) Python 3.11

```
FAIL: runTest (tests.config.TestControlledDict.runTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/zim-0.74.3/tests/config.py", line 256, in runTest
    self.assertTrue(mydict.modified)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: False is not true
```
Full build log: https://kojipkgs.fedoraproject.org//work/tasks/2658/88342658/build.log

**To Reproduce**
Steps to reproduce the behavior:
1. run `LANG=en_US.UTF-8 xvfb-run ./test.py` under Python 3.11

**Expected behavior**
Tests all pass.

**Environment (please complete the following information):**
 - Version [e.g. 0.74.0]: 0.74.3
 - OS: [e.g. linux/windows/mac os] Fedora 37 with Python 3.11
 - Language: en_US.UTF-8

--- a/zim/config/dicts.py
+++ b/zim/config/dicts.py
@@ -74,6 +74,13 @@ class ControlledDict(DefinitionOrderedDict, SignalEmitter, ConnectorMixin):
 			self.disconnect_from(v)
 		self.emit('changed')
 
+	def pop(self, key, default=None):
+		# Only emit changed once here
+		with self.block_signals('changed'):
+			v = DefinitionOrderedDict.pop(self, key, default)
+		self.emit('changed')
+		return v
+
 	def update(self, E=(), **F):
 		# Only emit changed once here
 		with self.block_signals('changed'):

